src/Nelmio/Alice/Instances/Populator/Methods/ArrayAdd.php

```diff
@@ -84,4 +85,37 @@ private function findAdderMethod($object, $property)
             return $method;
         }
     }
+
+    /**
+     * @param string $plural
+     *
+     * @return string
+     */
+    private function singularize($plural)
+    {
+        if (null === $this->singularizeClass || null === $this->singularizeMethod) {
+            // default to false so we only try this once
+            $this->singularizeClass = false;
```

Comment:
why does `singularizeClass` and `singularizeMethod` should be class properties? Couldn't they be regular variables?